logs/file.go

```diff
@@ -148,30 +148,42 @@ func (w *fileLogWriter) WriteMsg(when time.Time, msg string, level int) error {
 	if level > w.Level {
 		return nil
 	}
-	hd, d, h := formatTimeHeader(when)
-	msg = string(hd) + msg + "\n"
-	if w.Rotate {
-		w.RLock()
-		if w.needRotateHourly(len(msg), h) {
-			w.RUnlock()
-			w.Lock()
+
+	logFormat := "APACHE_FORMAT"
+	tempStruct := &AccessLogRecord{}
+	jsonDecodeError := json.Unmarshal([]byte(msg), tempStruct)
+	if jsonDecodeError == nil {
+		// If successfully unmarshelled the text, it was a JSON
+		// format, else it was an apache format
+		logFormat = "JSON_FORMAT"
+	}
+
+	if logFormat == "APACHE_FORMAT" {
+		hd, d, h := formatTimeHeader(when)
+		msg = string(hd) + msg + "\n"
+		if w.Rotate {
```

Comment:
it looks like that we don't do rotate if the format is not "APACHE_FORMAT"?

Reply:
I somehow thought rotating was only needed for `APACHE_FORMAT`. I'll fix that